### PR TITLE
Cleanup unused parts of concurrency.d

### PR DESCRIPTION
**Pull request by omerfirmak**
Since LocalRest now uses Channels, most of the old code is not used anymore.

### CODE REVIEW

File: source/geod24/concurrency.d [hunk 1]
```diff
@@ -93,7 +63,7 @@ private
         bool doesIt = false;
         static foreach (T; Types)
         {
-            static if (is(T == Tid))
+            static if (is(T : Channel!R, R))
```
Comment (Geod24) [inline source/geod24/concurrency.d]: Shouldn't we check `R` for local aliasing here ?

**Review by Geod24 — APPROVED** [review]
Me likey, just rebase

File: source/geod24/concurrency.d (hunk 1 above)
Comment (omerfirmak) [reply source/geod24/concurrency.d]: Done

**Comment by omerfirmak** [comment]
Rebased

File: source/geod24/concurrency.d [hunk 2]
```diff
@@ -107,8 +77,8 @@ private
 
     @safe unittest
     {
-        static struct Container { Tid t; }
-        static assert(!hasLocalAliasing!(Tid, Container, int));
+        static struct Container { Channel!int t; }
+        static assert(!hasLocalAliasing!(Channel!int, Container, int));
```
Comment (Geod24) [inline source/geod24/concurrency.d]:
```suggestion
        static assert(!hasLocalAliasing!(Channel!(Channel!int), Channel!int, Container, int));
        static assert( hasLocalAliasing!(Channel!(Channel!(int[]))));        
```

For testing the new code path ?

Comment (omerfirmak) [reply source/geod24/concurrency.d]: and it breaks immediately:)

Comment (Geod24) [reply source/geod24/concurrency.d]: We're passing locally aliased types around ?

Comment (omerfirmak) [reply source/geod24/concurrency.d]: `hasLocalAliasing!(Channel!(Channel!int))`

It didnt like the nested channels:)

```source/geod24/concurrency.d(66,24): Error: declaration R is already defined```

Comment (Geod24) [reply source/geod24/concurrency.d]: Hum, try replacing `R` with something else... E.g. `ChannelType` ?

Comment (Geod24) [reply source/geod24/concurrency.d]: https://issues.dlang.org/show_bug.cgi?id=20018

**Comment by Geod24** [comment]
Let's see how this fare

**Pull request merged**